plugins/delete-by-query/src/main/java/org/elasticsearch/action/deletebyquery/DeleteByQueryRequest.java

```diff
@@ -0,0 +1,264 @@
+        return this;
+    }
+
+    public DeleteByQueryRequest source(String querySource) {
+        this.source = new BytesArray(querySource);
+        return this;
+    }
+
+    public DeleteByQueryRequest source(byte[] querySource) {
+        return source(querySource, 0, querySource.length);
+    }
+
+    public DeleteByQueryRequest source(byte[] querySource, int offset, int length) {
+        return source(new BytesArray(querySource, offset, length));
+    }
+
+    public DeleteByQueryRequest source(BytesReference querySource) {
+        this.source = querySource;
+        return this;
+    }
+
+    public String routing() {
+        return this.routing;
+    }
+
+    public DeleteByQueryRequest routing(String routing) {
+        this.routing = routing;
+        return this;
+    }
+
+    public DeleteByQueryRequest routing(String... routings) {
+        this.routing = Strings.arrayToCommaDelimitedString(routings);
+        return this;
+    }
+
+    public DeleteByQueryRequest size(int size) {
+        this.size = size;
+        return this;
+    }
+
+    public Integer size() {
+        return size;
+    }
+
+
+    public Scroll scroll() {
+        return scroll;
+    }
+
+    public DeleteByQueryRequest scroll(Scroll scroll) {
+        this.scroll = scroll;
+        return this;
+    }
+
+    public DeleteByQueryRequest scroll(TimeValue keepAlive) {
+        return scroll(new Scroll(keepAlive));
+    }
+
+    public DeleteByQueryRequest scroll(String keepAlive) {
+        return scroll(new Scroll(TimeValue.parseTimeValue(keepAlive, null, getClass().getSimpleName() + ".keepAlive")));
+    }
+
+    public TimeValue timeout() {
+        return timeout != null ? new TimeValue(timeout) : null;
+    }
+
+    public DeleteByQueryRequest timeout(TimeValue timeout) {
+        this.timeout = timeout.millis();
+        return this;
+    }
+
+    public DeleteByQueryRequest timeout(String timeout) {
+        this.timeout = TimeValue.parseTimeValue(timeout, null, getClass().getSimpleName() + ".timeout").millis();
+        return this;
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        super.readFrom(in);
+        indices = in.readStringArray();
+        indicesOptions = IndicesOptions.readIndicesOptions(in);
+        types = in.readStringArray();
+        source = in.readBytesReference();
+        routing = in.readOptionalString();
+        size = in.readVInt();
+        if (in.readBoolean()) {
+            scroll = readScroll(in);
+        }
+        if (in.readBoolean()) {
+            timeout = in.readVLong();
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeStringArray(indices);
+        indicesOptions.writeIndicesOptions(out);
+        out.writeStringArray(types);
+        out.writeBytesReference(source);
+        out.writeOptionalString(routing);
+        out.writeVInt(size);
```

Comment:
Ok. `size` is initialized to 0 and can't be set to a negative value.

For my education, `writeVInt` does not support negative values at all or it's just a question of better serialization?